src/main/java/neqsim/thermodynamicoperations/ThermodynamicOperations.java

```diff
@@ -1184,7 +1184,7 @@ public void hydrateFormationTemperature(int structure) throws IsNaNException {
     }
 
     for (int i = 0; i < system.getPhase(4).getNumberOfComponents(); i++) {
-      ((ComponentHydrate) system.getPhases()[4].getComponent(i)).setHydrateStructure(structure - 1);
+      ((ComponentHydrate) system.getPhase(4).getComponent(i)).setHydrateStructure(structure - 1);
```

Comment:
In this case I beleive we have to used system.getPhases()[4] - as the hydrate phase is always number four (and getPhase(4) can point to a different phase.